src/engine/controls/enginecontrol.cpp

```diff
@@ -45,13 +47,14 @@ void EngineControl::setEngineBuffer(EngineBuffer* pEngineBuffer) {
     m_pEngineBuffer = pEngineBuffer;
 }
 
-void EngineControl::setCurrentSample(
-        const double dCurrentSample, const double dTotalSamples, const double dTrackSampleRate) {
-    SampleOfTrack sot;
-    sot.current = dCurrentSample;
-    sot.total = dTotalSamples;
-    sot.rate = dTrackSampleRate;
-    m_sampleOfTrack.setValue(sot);
+void EngineControl::setFrameInfo(mixxx::audio::FramePos currentPosition,
+        mixxx::audio::FramePos trackEndPosition,
+        mixxx::audio::SampleRate sampleRate) {
+    FrameInfo info;
```

Comment:
We should use direct initialization now that `FrameInfo` is no longer a POD. It now has an implicitly generated default ctor.